BlueprintUICommonControls/Sources/ScrollView.swift

```diff
@@ -260,8 +300,137 @@ fileprivate final class ScrollerWrapperView: UIView {
                 self.scrollView.contentOffset.x = -contentInset.left
             }
         }
+    }
+    
+    //
+    // MARK: UIView
+    //
+    
+    public override func didMoveToWindow() {
+        super.didMoveToWindow()
+        
+        if self.window != nil {
+            self.updateBottomContentInsetWithKeyboardFrame()
+        }
+    }
+    
+    public override func didMoveToSuperview() {
+        super.didMoveToSuperview()
+        
+        if self.superview != nil {
+            self.updateBottomContentInsetWithKeyboardFrame()
+        }
+    }
+}
 
 
+extension ScrollView
+{
+    // Calculates the correct content inset to apply for the given inputs.
+    
+    static func finalContentInset(
```

Comment:
Why is this static? What's "final" about it? If it's for calculating the content inset, maybe a name like `calculateContentInset()`?

Reply:
Static since it makes testing easier – just pass in the needed values and verify the output.

Reply:
Renamed to `calculateContentInset`.